bones/generate.go

```diff
@@ -0,0 +1,113 @@
+package bones
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func Generate(protoSet, methodDir string, force bool, targets []string) error {
+	files, err := loadProtoSet(protoSet)
+	if err != nil {
+		return err
+	}
+
+	files.RangeFiles(func(fd protoreflect.FileDescriptor) bool {
+		err = genFile(fd, methodDir, force, targets)
+		return err == nil
+	})
+
+	return nil
+}
+
+func loadProtoSet(protoSet string) (*protoregistry.Files, error) {
+	b, err := os.ReadFile(protoSet)
+	if err != nil {
+		return nil, err
+	}
+	fds := &descriptorpb.FileDescriptorSet{}
+	if err := proto.Unmarshal(b, fds); err != nil {
+		return nil, err
+	}
+	files, err := protodesc.NewFiles(fds)
```

Comment:
is the protoregistry really needed at this stage? does it give us anything that iterating the FDs wouldn't give us?

Reply:
(not that i'm against it, just curious).

Reply:
seemed like it would be more involved. no strong reason to use protoregistry except it solved the problem easily.

Reply:
Oh! I didn't realise these were different filedescriptors. cheers.